File: sdks/csharp/rest/src/Brume.Rest/Stats/Requests/AnalyticsRequest.cs
using Brume.Rest.Core;
using global::System.Text.Json.Serialization;

namespace Brume.Rest;

[Serializable]
public record AnalyticsRequest
{
    /// <summary>
    /// Time window in seconds. Default 3600 (1h). Clamped to [300, 21600] (5 min to 6 h).
    /// </summary>
    [JsonIgnore]
    public long? WindowSecs { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/rest/src/Stats/Requests/AnalyticsRequest.php
<?php

namespace Brume\Rest\Stats\Requests;

use Brume\Rest\Core\Json\JsonSerializableType;

class AnalyticsRequest extends JsonSerializableType
{
    /**
     * @var ?int $windowSecs Time window in seconds. Default 3600 (1h). Clamped to [300, 21600] (5 min to 6 h).
     */
    public ?int $windowSecs;

    /**
     * @param array{
     *   windowSecs?: ?int,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->windowSecs = $values['windowSecs'] ?? null;
    }
}
